app/views/govuk_publishing_components/components/_search.html.erb

```diff
@@ -3,6 +3,8 @@
   class_name = "gem-c-search--on-govuk-blue" if local_assigns[:on_govuk_blue].eql?(true)
   size ||= ""
   no_margin ||= false
+  no_border ||= false
+  class_name = "#{class_name} gem-c-search--no-border" if no_border
```

Comment:
I can't tell if I'm reading this right or not, but it seems like each of these options is mutually exclusive i.e. it's not possible to have the component be both e.g. large and without a border. Do we not need this behaviour?

Reply:
I'm just looking at this currently and the options are not mutually exclusive. 
`class_name` builds on itself by concatenating the existing `class_name` and the new class if the conditions are met. So it is possible to have the component large, and with a border, and with no margin etc.

However, this seems like a slightly confusing way to achieve that outcome. We can refactor this to use a word array instead, which should improve readability.